### PR TITLE
Add renovate config and automated GPU driver updates for CUDA

### PR DESCRIPTION
**Pull request by ganeshkumarashok**
This adds renovate config for automated updates via the renovate bot.

This configuration will auto-update CUDA drivers to the latest production branch via creating new PRs. It uses the Nvidia releases page as the source: https://docs.nvidia.com/datacenter/tesla/drivers/releases.json.

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "customManagers": [
+    {
+      "customType": "regex",
+      "fileMatch": [
+        "driver_config.yml"
+      ],
+      "matchStrings": [
+        "cuda:\\s*(?:#.*)?\\n\\s*version:\\s*\"(?<currentValue>[^\"]+)\""
```
Comment (djsly) [inline renovate.json:10]: why not match for renovate ? right now the comment in the file is ignored

File: renovate.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      ],
+      "depNameTemplate": "nvidia-cuda-driver",
```
Comment (djsly) [inline renovate.json:12]: this is hardcoded and not extracted

File: renovate.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      "datasourceTemplate": "custom.nvidia-driver",
```
Comment (djsly) [inline renovate.json:13]: hardcoded, not extracted

File: renovate.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      "versioningTemplate": "loose"
```
Comment (djsly) [inline renovate.json:14]: hardcoded, not extracted

File: renovate.json [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  ],
+  "customDatasources": {
+  "nvidia-driver": {
+    "defaultRegistryUrlTemplate": "https://docs.nvidia.com/datacenter/tesla/drivers/releases.json",
+    "transformTemplates": [
+      "{ \"releases\": $map($reduce($map($keys($), function($k) { { \"key\": $k, \"value\": $lookup($, $k) } }), function($acc, $v) { $append($acc, $v.value.driver_info) }, []), function($info) { { \"version\": $info.release_version } }) }"
+    ]
+  }
+ },
+  "packageRules": [
```
Comment (djsly) [inline renovate.json:25]: what is this packageRule for ?

File: renovate.json [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    {
+      "excludePackagePatterns": [".*"],
+      "managers": ["regex"],
+      "matchManagers": ["regex"],
+      "enabled": true
+    }
+  ],
+  "logLevelRemap": [
```
Comment (djsly) [inline renovate.json:33]: do we want to keep this around ?

**Review by djsly — CHANGES_REQUESTED** [review]

File: driver_config.yml [hunk 7]
```diff
@@ -1,4 +1,5 @@
 cuda:
+  # renovate: datasource=custom.nvidia-driver depName=nvidia-cuda-driver
```
Comment (djsly) [inline driver_config.yml:2]: # renovate: datasource=custom.nvidia-driver depName=nvidia-cuda-driver versioning:loose

File: renovate.json (hunk 1 above)
Comment (ganeshkumarashok) [reply renovate.json]: Thanks, updated.

File: renovate.json (hunk 3 above)
Comment (ganeshkumarashok) [reply renovate.json]: Extracting it now

File: renovate.json (hunk 5 above)
Comment (ganeshkumarashok) [reply renovate.json]: Had it to see about reducing the files it was tracking - unnecessary now

File: driver_config.yml (hunk 7 above)
Comment (ganeshkumarashok) [reply driver_config.yml]: versioning=loose right?

File: renovate.json (hunk 6 above)
Comment (ganeshkumarashok) [reply renovate.json:22]: I think keeping till we see regular updates for cuda drivers would be helpful. Now, this repo hasn't got approval to use renovate yet so it's not been fully tested.

File: driver_config.yml (hunk 7 above)
Comment (djsly) [reply driver_config.yml]: looks like they are using semver no ?

Comment (ganeshkumarashok) [reply driver_config.yml]: Renovate did not parse it properly when I had it set to semver. Checked https://docs.renovatebot.com/modules/versioning/loose/ and https://docs.renovatebot.com/modules/versioning/semver/. Nvidia drivers don't follow strict semver 2.0 guidelines like rule 8, for instance: https://semver.org/.

Haven't checked renovate semver validation code though, but the docs do mention that following semver 2.0 is required.

Comment (djsly) [reply driver_config.yml]: the default is: https://docs.renovatebot.com/modules/versioning/semver-coerced/. which is supposed to be forgiving

**Review by djsly — APPROVED** [review]

**Pull request merged**